components/eam/src/physics/cam/modal_aer_opt.F90

```diff
@@ -1214,242 +1214,280 @@ subroutine modal_aero_sw(list_idx, dt, state, pbuf, nnite, idxnite, is_cmip6_vol
    integer :: istat
 
-   real(r8), pointer :: dgnumwet(:,:)  ! wet number mode diameter (m)
-   real(r8), pointer :: qaerwat(:,:)   ! aerosol water (g/g)
+   real(r8), pointer :: dgnumwet(:,:)  ! wet number mode diameter [m]
+   real(r8), pointer :: qaerwat(:,:)   ! aerosol water [g/g]
 
    real(r8) :: sigma_logr_aer          ! geometric standard deviation of number distribution
    real(r8) :: alnsg_amode
-   real(r8) :: xrad(pcols)
-   real(r8) :: cheby(ncoef,pcols,pver)  ! chebychef polynomials
+   real(r8) :: cheby(ncoef,pcols,pver) ! chebychef polynomials
+   real(r8) :: radsurf(pcols,pver)     ! aerosol surface mode radius
+   real(r8) :: logradsurf(pcols,pver)  ! log(aerosol surface mode radius)
 
    real(r8) :: mass(pcols,pver) ! layer mass
 
-   real(r8),    pointer :: specmmr(:,:)        ! species mass mixing ratio
-   real(r8)             :: specdens            ! species density (kg/m3)
-   complex(r8), pointer :: specrefindex(:)     ! species refractive index
+   real(r8),    pointer :: specmmr(:,:)        ! species mass mixing ratio [g/g]
+   real(r8),allocatable :: volf(:,:)           ! volume fraction of insoluble aerosol
+   real(r8),allocatable :: specdens(:)         ! species density for all species [kg/m3]
+   complex(r8),allocatable :: specrefindex(:,:)     ! species refractive index
 
-   real(r8) :: vol(pcols)       ! volume concentration of aerosol specie (m3/kg)
-   real(r8) :: dryvol(pcols)    ! volume concentration of aerosol mode (m3/kg)
-   real(r8) :: wetvol(pcols)    ! volume concentration of wet mode (m3/kg)
-   real(r8) :: watervol(pcols)  ! volume concentration of water in each mode (m3/kg)
+   real(r8) :: dryvol(pcols)    ! volume concentration of aerosol mode [m3/kg]
+   real(r8) :: wetvol(pcols)    ! volume concentration of wet mode [m3/kg]
+   real(r8) :: watervol(pcols)  ! volume concentration of water in each mode [m3/kg]
    real(r8) :: refr(pcols)      ! real part of refractive index
    real(r8) :: refi(pcols)      ! imaginary part of refractive index
    complex(r8) :: crefin(pcols) ! complex refractive index
-   real(r8), pointer :: refrtablw(:,:) ! table of real refractive indices for aerosols
-   real(r8), pointer :: refitablw(:,:) ! table of imag refractive indices for aerosols
-   real(r8), pointer :: absplw(:,:,:,:) ! specific absorption
+   real(r8), pointer :: state_q(:,:,:)  ! state%q
 
    integer  :: itab(pcols), jtab(pcols)
    real(r8) :: ttab(pcols), utab(pcols)
    real(r8) :: cabs(pcols,ncoef)
-   real(r8) :: pabs(pcols)      ! parameterized specific absorption (m2/kg)
-   real(r8) :: dopaer(pcols)    ! aerosol optical depth in layer
+   real(r8) :: pabs      ! parameterized specific absorption [m2/kg]
+   real(r8) :: dopaer    ! aerosol optical depth in layer
 
-   integer, parameter :: nerrmax_dopaer=1000
    integer  :: nerr_dopaer = 0
-   real(r8) :: volf             ! volume fraction of insoluble aerosol
 
    character(len=*), parameter :: subname = 'modal_aero_lw'
```

Comment:
We can remove this `subname`. It was used to tell the user that the code blew up in `subname` subroutine. Since you have refactored it now, it doesn't hold any meaning.